### PR TITLE
Add support for new SMTP config schema

### PR DESCRIPTION
**Pull request by jaclarke**
Adds support for the new SMTP config introduced in https://github.com/edgedb/edgedb/pull/7942.

image

@scotttrinh I'm not sure how the cloud magic smtp provider will be introspected, is it going to be a another type extending `cfg::EmailProvider` that won't be user configurable, or is it another `SMTPProviderConfig` which we have to make readonly in the UI?

~Also there's a new potential tripping hazard when updating an existing smtp provider, as I think we now have to update the whole config object, so that will reset the password every time unless you re-enter the password whenever you make a change.~
~Disabled editing of smtp configs for now.~
Re-enabled smtp config editing, but with a warning message if you don't re-enter the password.

### CODE REVIEW

File: shared/studio/tabs/auth/providers.tsx [hunk 1]
```diff
@@ -50,11 +51,28 @@ export const ProvidersTab = observer(function ProvidersTab() {
...
-              {state.providers.map((provider) => (
-                <ProviderCard key={provider.name} provider={provider} />
-              ))}
-            </div>
+            <>
+              {state.noEmailProviderWarning ? (
+                <div className={styles.emailProviderWarning}>
+                  <WarningIcon />
+                  <span>Warning:</span> You have enabled auth providers
+                  requiring email, but no SMTP provider is configured.
```
Comment (scotttrinh) [inline shared/studio/tabs/auth/providers.tsx:59]: It's a bit more complicated than this since you might avoid configuring SMTP and just use webhooks.

Comment (scotttrinh) [reply shared/studio/tabs/auth/providers.tsx:59]: Now that I think of it, we should have an opt-out option which sets the pointer to `{}` without losing your config.

Comment (jaclarke) [reply shared/studio/tabs/auth/providers.tsx]: Ok, I've updated the warnings to be more specific and also take into account webhook config. And added a button deselect the currently enabled smtp provider.

File: shared/studio/tabs/auth/smtp.tsx [hunk 2]
```diff
@@ -191,26 +242,30 @@ const EmailProviderCard = observer(function EmailProviderCard({
...
           </div>
 
+          {draftState.formChanged &&
+          draftState.getConfigValue("password").trim() === "" ? (
+            <div className={styles.passwordWarning}>
+              <WarningIcon />
+              <div>
+                <span>Warning:</span> You have not entered a password. If there
+                is an existing password, it will be deleted when you update
+                this SMTP provider unless you re-enter the password above.
```
Comment (scotttrinh) [inline shared/studio/tabs/auth/smtp.tsx:255]: Yeah, this seems fine!

File: shared/studio/tabs/auth/state/index.tsx [hunk 3]
```diff
@@ -392,19 +392,57 @@ export class AuthAdminState extends Model({
...
+    const magicLinkWebhook = this.webhooks?.some((wh) =>
+      wh.events.includes("MagicLinkRequested")
     );
+
+    const smtpConfigured = this.emailProviders?.some(
+      (provider) => provider.name === this.currentEmailProvider
+    );
+
+    return {
+      verificationNoSmtp: emailVerification && !smtpConfigured,
```
Comment (scotttrinh) [inline shared/studio/tabs/auth/state/index.tsx:440]: verification can be a webhook as well: `EmailVerificationRequested`

https://github.com/edgedb/edgedb/blob/82fac253798a4bdccae2ebcca9aa14e3a7d0ac4d/edb/lib/ext/auth.edgeql#L376

**Review by scotttrinh — APPROVED** [review]

**Pull request merged**